great_expectations/dataset/sqlalchemy_dataset.py

```diff
@@ -1242,11 +1242,19 @@ def create_temporary_table(self, table_name, custom_sql, schema_name=None):
         # similar, for example.
         ###
 
-        if self.sql_engine_dialect.name.lower() == "bigquery":
+        engine_dialect = self.sql_engine_dialect.name.lower()
+        if type(engine_dialect) != str:
+            engine_dialect = str(engine_dialect, "utf-8")
+
+        if engine_dialect == "bigquery":
```

Comment:
```suggestion
        engine_dialect = self.sql_engine_dialect.name.lower()
        # handle cases where dialect.name.lower() returns a byte string (e.g. databricks)
        if isinstance(engine_dialect, bytes):
            engine_dialect = str(engine_dialect, "utf-8")

        if engine_dialect == "bigquery":
```

Stylistically, we prefer using `isinstance` to check types, vs. the `type(x) == y` or `type(x) != y` pattern.

Reply:
Also suggest adding comment for clarity.